From 41d67213607e7ff20b7a3ca833f1cfde9780da65 Mon Sep 17 00:00:00 2001
From: Hermet Park <hermet@lottiefiles.com>
Date: Sat, 7 Sep 2024 01:35:09 +0900
Subject: [PATCH] renderer: ++reliability in text drawing

Allow the canvas to pass through
even if text elements are not properly supported.

issue: https://github.com/thorvg/thorvg/issues/2715
---
 src/renderer/tvgText.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/thirdparty/thorvg/src/renderer/tvgText.h b/thirdparty/thorvg/src/renderer/tvgText.h
index 746b85bea6..55d33ffd4b 100644
--- a/thirdparty/thorvg/src/renderer/tvgText.h
+++ b/thirdparty/thorvg/src/renderer/tvgText.h
@@ -89,6 +89,7 @@ struct Text::Impl
 
     bool render(RenderMethod* renderer)
     {
+        if (!loader) return true;
         renderer->blend(paint->blend(), true);
         return PP(shape)->render(renderer);
     }
